plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeLocalConcurrentWritesTest.java

```diff
@@ -601,6 +603,442 @@ private void testConcurrentInsertsReconciliationFailure(boolean partitioned)
+                            getQueryRunner().execute("TRUNCATE TABLE " + tableName);
+                            return true;
+                        }
+                        catch (Exception e) {
+                            RuntimeException trinoException = getTrinoExceptionCause(e);
+                            try {
+                                assertThat(trinoException).hasMessage("Failed to write Delta Lake transaction log entry");
+                            }
+                            catch (Throwable verifyFailure) {
+                                if (verifyFailure != e) {
+                                    verifyFailure.addSuppressed(e);
+                                }
+                                throw verifyFailure;
+                            }
+                            return false;
+                        }
+                    }))
+                    .collect(toImmutableList());
+
+            long successfulTruncatesCount = futures.stream()
+                    .map(future -> tryGetFutureValue(future, 20, SECONDS).orElseThrow(() -> new RuntimeException("Wait timed out")))
+                    .filter(success -> success)
+                    .count();
+
+            assertThat(successfulTruncatesCount).isGreaterThanOrEqualTo(1);
+            assertThat(query("SELECT * FROM " + tableName)).returnsEmptyResult();
+            assertQuery(
+                    "SELECT version, operation, isolation_level, is_blind_append FROM \"" + tableName + "$history\"",
+                    "VALUES (0, 'CREATE TABLE AS SELECT', 'WriteSerializable', true)" +
+                            LongStream.rangeClosed(1, successfulTruncatesCount)
+                                    .boxed()
+                                    .map("(%s, 'TRUNCATE', 'WriteSerializable', false)"::formatted)
+                                    .collect(joining(", ", ", ", "")));
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+            executor.shutdownNow();
+            assertTrue(executor.awaitTermination(10, SECONDS));
+        }
+    }
+
+    @Test
+    public void testConcurrentDeletePushdownAndBlindInsertsReconciliation()
+            throws Exception
+    {
+        int threads = 3;
+        CyclicBarrier barrier = new CyclicBarrier(threads);
+        ExecutorService executor = newFixedThreadPool(threads);
+        String tableName = "test_concurrent_delete_and_inserts_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a, part)  WITH (partitioned_by = ARRAY['part']) AS VALUES (1, 10), (11, 20)", 2);
+
+        try {
+            // delete whole data from the table while concurrently adding new blind inserts
+            executor.invokeAll(ImmutableList.<Callable<Void>>builder()
+                            .add(() -> {
+                                barrier.await(10, SECONDS);
+                                getQueryRunner().execute("DELETE FROM " + tableName);
```

Comment:
Could you add scenario with where clause?
Something like:
DELETE FROM table WHERE a > 10;
INSERT INTO table VALUES (8, 20);
INSERT INTO table VALUES (21, 30);

Reply:
Added `testConcurrentDeleteAndBlindInsertsReconciliation`